express-notes-api/server.js

```diff
@@ -0,0 +1,118 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+
+const app = express();
+app.use(express.json());
+
+app.get('/api/notes', async (req, res) => {
+  const array = [];
+  const data = await readJSONData();
+  const notesList = data.notes;
+  for (const entries in notesList) {
+    array.push(notesList[entries]);
+  }
+  res.json(array);
+});
+
+app.get('/api/notes/:id', async (req, res) => {
+  const { id } = req.params;
+  const data = await readJSONData();
+  const notesList = data.notes;
+  if (id <= 0) {
+    res.json({ error: 'Please enter a positive number.' });
+  } else if (notesList[id]) {
+    res.json(notesList[id]);
+  } else {
+    res.status(404).json({ error: `Entry Id: ${id} does not exist.` });
+  }
+});
+
+app.post('/api/notes', async (req, res) => {
+  try {
+    const newEntry = req.body;
+    const data = await readJSONData();
+    if (!newEntry.content) {
+      res.status(400).json({ error: 'Content is a required field.' });
+    } else if (newEntry.content) {
+      await createNotes(data, newEntry);
+      res.json(newEntry);
+    } else {
+      throw new Error('An unexpected error occurred.');
+    }
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ error: `${error}` });
+  }
+});
+
+app.delete('/api/notes/:id', async (req, res) => {
+  const { id } = req.params;
+  const data = await readJSONData();
+  deleteNotes(data, id, res);
+});
+
+async function readJSONData() {
+  const contents = await readFile('data.json');
+  const data = JSON.parse(contents);
+  return data;
+}
+
+async function createNotes(data, newEntry) {
+  newEntry.id = data.nextId++;
+  data.notes[newEntry.id] = newEntry;
+  data = JSON.stringify(data, null, 2);
+  await writeFile('data.json', data);
+}
+
+async function deleteNotes(data, id, res) {
+  if (isNaN(id) || id <= 0) {
+    res.status(400).json({ error: 'Please enter a positive number.' });
+    return;
+  } else if (!data.notes[id]) {
+    res.status(404).json({ error: `Entry Id: ${id} does not exist.` });
+    return;
+  } else {
+    delete data.notes[id];
+  }
+
+  try {
+    data = JSON.stringify(data, null, 2);
+    await writeFile('data.json', data);
```

Comment:
You repeat these two lines of code several times. It would be good to put them in a separate function.